src/pages/singleVideoPage/singleVidPage.jsx

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import { Footer, Nav, Sidebar } from "../../components";
+import ReactPlayer from "react-player";
+import { useSingleVidContext } from "../../context/singleVidContext";
+import { useLikedVideoContext } from "../../context/likedVideoContext";
+import { useWatchLaterContext } from "../../context/watchLaterContext";
+import { useState } from "react";
+
+function SingleVidPage() {
+  const { singleVidState } = useSingleVidContext();
+  const { likedVideoState, likedVideoDispatch } = useLikedVideoContext();
+  const { watchLaterState, watchLaterDispatch } = useWatchLaterContext();
+
+  function watchLaterToggle(item) {
+    if (
+      watchLaterState.watchLaterList.find((vid) => vid._id === item._id) !==
+      undefined
+    ) {
+      return watchLaterDispatch({
+        type: "REMOVE_LATER_VIDEO",
+        payload: singleVidState.vidItem._id,
+      });
+    } else {
+      return watchLaterDispatch({
+        type: "ADD_LATER_VIDEO",
+        payload: singleVidState.vidItem,
+      });
+    }
+  }
+
+  function likeToggle(item) {
+    if (
+      likedVideoState.likedVidList.find((vid) => vid._id === item._id) !==
+      undefined
+    ) {
+      return likedVideoDispatch({
+        type: "REMOVE_LIKED_VIDEO",
+        payload: singleVidState.vidItem._id,
+      });
+    } else {
+      return likedVideoDispatch({
+        type: "ADD_LIKED_VIDEO",
+        payload: singleVidState.vidItem,
+      });
+    }
+  }
+
+  return (
+    <div>
```

Comment:
Suggestion: You can use React Fragments here instead of div if it is only used for grouping code together as adding extra divs to DOM is not efficient